packages/backend/src/machine-utils.spec.ts

```diff
@@ -84,3 +84,77 @@ test('Check isPodmanMachineRootful functionality', async () => {
   spyReadFile.mockResolvedValue(JSON.stringify({ Rootful: true }));
   await expect(machineUtils.isPodmanMachineRootful()).resolves.toBe(true);
 });
+
+test('Fail isPodmanMachineRootful functionality be false if Rootful does not exist in the root of the object, or in HostUser', async () => {
+  const fakeMachineInfoJSON = {
+    Host: {
+      Arch: 'amd64',
+      CurrentMachine: '',
+      DefaultMachine: '',
+      EventsDir: 'dir1',
+      MachineConfigDir: 'dir2',
+      MachineImageDir: 'dir3',
+      MachineState: '',
+      NumberOfMachines: 5,
+      OS: 'windows',
+      VMType: 'wsl',
+    },
+  };
+
+  vi.spyOn(extensionApi.process, 'exec').mockImplementation(
+    () =>
+      new Promise<extensionApi.RunResult>(resolve => {
+        resolve({ stdout: JSON.stringify(fakeMachineInfoJSON) } as extensionApi.RunResult);
+      }),
+  );
+
+  // Mock existsSync to return true (the "fake" file is there)
+  vi.mock('node:fs');
+  vi.spyOn(fs, 'existsSync').mockImplementation(() => {
+    return true;
+  });
+
+  // Mock the readFile function to return the "fake" file with Rootful not existing
+  const spyReadFile = vi.spyOn(fs.promises, 'readFile');
+
+  // Mock reading the file to have Rootful as true
+  spyReadFile.mockResolvedValue(JSON.stringify({}));
+  await expect(machineUtils.isPodmanMachineRootful()).resolves.toBe(false);
+});
+
+test('Pass true if Rootful is in HostUser', async () => {
+  const fakeMachineInfoJSON = {
+    Host: {
+      Arch: 'amd64',
+      CurrentMachine: '',
+      DefaultMachine: '',
+      EventsDir: 'dir1',
+      MachineConfigDir: 'dir2',
+      MachineImageDir: 'dir3',
+      MachineState: '',
+      NumberOfMachines: 5,
+      OS: 'windows',
+      VMType: 'wsl',
+    },
+  };
+
+  vi.spyOn(extensionApi.process, 'exec').mockImplementation(
+    () =>
+      new Promise<extensionApi.RunResult>(resolve => {
+        resolve({ stdout: JSON.stringify(fakeMachineInfoJSON) } as extensionApi.RunResult);
+      }),
+  );
+
+  // Mock existsSync to return true (the "fake" file is there)
+  vi.mock('node:fs');
+  vi.spyOn(fs, 'existsSync').mockImplementation(() => {
+    return true;
+  });
+
+  // Mock the readFile function to return the "fake" file with Rootful not existing
+  const spyReadFile = vi.spyOn(fs.promises, 'readFile');
+
+  // Mock reading the file to have Rootful as true
+  spyReadFile.mockResolvedValue(JSON.stringify({ HostUser: { Rootful: true } }));
```

Comment:
Why not put rootful in the fake info above? (Or can we mock readMachineConfig() directly to just return the JSON here?)

Reply:
See below:

```json
{
  "Host": {
    "Arch": "arm64",
    "CurrentMachine": "podman-machine-default",
    "DefaultMachine": "",
    "EventsDir": "/var/folders/cn/pm5mgrcs5h339nphypxtvd080000gn/T/storage-run-501/podman",
    "MachineConfigDir": "/Users/cdrage/.config/containers/podman/machine/applehv",
    "MachineImageDir": "/Users/cdrage/.local/share/containers/podman/machine/applehv",
    "MachineState": "Running",
    "NumberOfMachines": 1,
    "OS": "darwin",
    "VMType": "applehv"
  },
  "Version": {
    "APIVersion": "5.0.0-dev",
    "Version": "5.0.0-dev",
    "GoVersion": "go1.21.6",
    "GitCommit": "0ace66f1aed453ebc25febbfc79636095d08ce78",
    "BuiltTime": "Fri Mar  1 13:46:53 2024",
    "Built": 1709318813,
    "OsArch": "darwin/arm64",
    "Os": "darwin"
  }
}
```

For podman machine, we use `podman machine info --format json` which gets the directory of where the actual json information is. 

That's why we fake it as the isPodmanRootful command first gets that.